src/fragments/lib/datastore/native_common/advanced-workflows.mdx

```diff
@@ -0,0 +1,272 @@
+## Custom Primary Keys
+
+By default, all DataStore models have an `id` field by default that is automatically populated on the client with a UUID v4, allowing DataStore to generate non-colliding globally unique identifiers in a scalable way.
+
+While UUIDs have desirable properties (they are large, non-sequential and opaque), there are times when a custom primary key is needed. For instance, to:
+
+
+* Have friendly/readable identifiers (surrogate/opaque vs. natural keys)
+* Define composite primary keys
+* Customize data partitioning to optimize for scale (especially important when planning to handle large amounts of data in short periods of time)
+* Selectively synchronize data to clients (e.g. by fields like `deviceId`, `userId` or similar)
+* Prioritize the sort order in which objects are returned by the sync queries
+* Make existing data consumable and syncable by DataStore
+
+### Schema
+
+A schema with the typical `id` field looks like this:
+
+```graphql
+type Todo4 @model {
+	id: ID!
+	name: String!
+	description: String
+}
+```
+
+With custom primary keys, you can provide a unique value using the `@primaryKey` directive:
+
+```graphql
+type Todo @model {
+	customId: ID! @primaryKey
+	name: String!
+	description: String
+}
+```
+
+With sort key:
```

Comment:
Consider having these inside CodeBlock switchers so customers don't have to scroll as much to see all examples and can easily flip between the examples.

maybe these could be good tab headings:
|Basic custom identifier (hash key)| Hash key and sort key| Custom query field|

Reply:
I tried this, and it feels less straightfoward to me, and a bit unnatural / harder to parse. What if we compared 1) a schema without CPK, 2) a schema with CPK, and then had a separate section (later) on composite keys?